scipy/linalg/_solvers.py

```diff
@@ -181,7 +181,7 @@ def solve_continuous_lyapunov(a, q):
     # Call the Sylvester equation solver
     trsyl = get_lapack_funcs('trsyl', (r, f))
 
-    dtype_string = 'T' if r_or_c == float else 'C'
+    dtype_string = 'T' if r_or_c is float else 'C'
```

Comment:
actually, I think these should be `isinstance(r_or_c, float)`.

Reply:
well, not quite, since `r_or_c` is a type, not an object. I think this is right, but some others aren't.